be/src/util/numeric_types.h

```diff
@@ -0,0 +1,58 @@
+// Copyright 2021-present StarRocks, Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#pragma once
+
+#include <limits>
+#include <type_traits>
+
+namespace starrocks {
+
+template <typename FromType, typename ToType>
+static constexpr FromType floating_to_intergral_lower_bound =
+        static_cast<FromType>(std::numeric_limits<ToType>::lowest());
+
+template <typename FromType, typename ToType>
+static constexpr FromType floating_to_intergral_upper_bound = static_cast<FromType>(2) *
+                                                              (std::numeric_limits<ToType>::max() / 2 + 1);
+
+/// Check whether the value of type `FromType` overflows when converted to type `ToType`.
+/// If overflow, return true; otherwise, return false.
+template <typename FromType, typename ToType>
+bool check_number_overflow(FromType value) {
```

Comment:
NIT: It will be better to create UT for this function, easy to construct and cover corner cases. And it won't be influenced by the usage of this function.